_episodes/02-software-development.md

```diff
@@ -19,24 +17,137 @@ View (user interface), and Controller (input/output and data manipulation)."
+![copy-template-repository](../fig/copy-template-repository.png)
+4. Click the `Create repository from template` button and wait for GitHub to import the copy of the repository under your account.
+5. At this point GitHub may ask you to authenticate. If this happens and
+you do not have 2-Factor-Authentication (2FA) enabled in your
+GitHub account, you can just enter your password to proceed. If you are using 2FA, you may get a message:
+"Your old project requires credentials for read-only access. We will only temporarily store them for importing." and
+should use a pre-generated personal access token as your password here.
+6. Locate the copied repository under your own GitHub account.
+![github-template-repository](../fig/own-template-repository.png)
+
+> ## Obtain the Software Project Locally
+> Using a command line shell, clone the copied repository from your GitHub account  into your computer.
+> Which command(s) would you use to get a detailed list of contents of the directory you have just cloned?
+> > ## Solution
+> > 1. Find the URL of the software project repository to clone from your GitHub account. Make sure you do not clone the
+>original template repository but rather your own copy, as you should be able to push commits to it later on.
+> > 2. Do:
+> > `git clone https://github.com/<YOUR_GITHUB_USERNAME>/python-intermediate-inflammation`
+> > 3. Navigate into the cloned repository in your command line shell:
+> > `cd python-intermediate-inflammation`
+> > 4. List the contents of the directory:
+> > `ls -l`
+> > Remember the `-l` flag of the `ls` command and also how to get help for commands in shell: `man ls`.
+> {: .solution}
+>
+{: .challenge}
+
+### Our Software Project Structure
+Let’s inspect the content of the software project from the command line. From the root directory of the project, you can
+ use the command `ls -l` to get a more detailed list of the contents. You should see something similar to the following.
+
+~~~
+$ ls -l
+total 24
+-rw-r--r--   1 carpentry  users  1055 20 Apr 15:41 README.md
+drwxr-xr-x  18 carpentry  users   576 20 Apr 15:41 data
+drwxr-xr-x   5 carpentry  users   160 20 Apr 15:41 inflammation
+-rw-r--r--   1 carpentry  users  1122 20 Apr 15:41 patientdb.py
+drwxr-xr-x   4 carpentry  users   128 20 Apr 15:41 tests
+~~~
+{: .language-bash}
+
+As can be seen from the above, our software project contains the `README` file (that typically describes the project, 
+its usage, installation, authors and how to contribute), Python script `patientdb.py`, 
+and three directories - `inflammation`, `data` and `tests`. The Python script `patientdb.py` provides the main
+entry point in the application - we will have a more detailed look into it shortly. 
+On closer inspection, we can see that the `inflammation` directory contains two more Python scripts - 
+`view.py` and `model.py`.
```

Comment:
Feels like a comment similar to "we'll look at these later" is needed here too. Could rephrase as "The Python script `patientdb.py` provides the main
entry point in the application, and on closer inspection, we can see that the `inflammation` directory contains two more Python scripts - `view.py` and `model.py`.  We will have a more detailed look into these shortly."

Reply:
Addressed via 7e7e63c.